src/m365/teams/commands/conversationmember/conversationmember-add.ts

```diff
@@ -0,0 +1,282 @@
+            headers: {
+              'content-type': 'application/json;odata=nometadata',
+              'accept': 'application/json;odata.metadata=none'
+            },
+            responseType: 'json',
+            data: {
+              '@odata.type': '#microsoft.graph.aadUserConversationMember',
+              'roles': roles,
+              'user@odata.bind': `https://graph.microsoft.com/v1.0/users('${userId}')`
+            }
+          };
+          return request.post(requestOptions);
+        }
+        for(const userId of userIds) {
+          tasks.push(addUser(userId));
+        }
+        return Promise.all(tasks);
+      }).then((): void => {
+        if (this.verbose) {
+          logger.logToStderr(chalk.green('DONE'));
+        }
+
+        cb();
+      }, (err: any): void => this.handleError(err, logger, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-i, --teamId [teamId]'
+      },
+      {
+        option: '--teamName [teamName]'
+      },
+      {
+        option: '-c, --channelId [channelId]'
+      },
+      {
+        option: '--channelName [channelName]'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userDisplayName [userDisplayName]'
+      },
+      {
+        option: '--owner'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (args.options.teamId && args.options.teamName) {
+      return 'Specify either teamId or teamName, but not both.';
+    }
+
+    if (!args.options.teamId && !args.options.teamName) {
+      return 'Specify teamId or teamName, one is required';
+    }
+
+    if (args.options.teamId && !Utils.isValidGuid(args.options.teamId as string)) {
+      return `${args.options.teamId} is not a valid GUID`;
+    }
+
+    if (args.options.channelId && args.options.channelName) {
+      return 'Specify either channelId or channelName, but not both.';
+    }
+
+    if (!args.options.channelId && !args.options.channelName) {
+      return 'Specify channelId or channelName, one is required';
+    }
+
+    if (args.options.channelId && !Utils.isValidTeamsChannelId(args.options.channelId as string)) {
+      return `${args.options.channelId} is not a valid Teams ChannelId`;
+    }
+
+    if (args.options.userId && args.options.userDisplayName) {
+      return 'Specify either userId or userDisplayName, but not both.';
+    }
+
+    if (!args.options.userId && !args.options.userDisplayName) {
+      return 'Specify userId or userDisplayName, one is required';
+    }
+
+    return true;
+  }
+
+  private getTeamId(args: CommandArgs): Promise<string> {
+    if (args.options.teamId) {
+      return Promise.resolve(args.options.teamId);
+    }
+
+    return new Promise<string>((resolve: (channelId: string) => void, reject: (error: any) => void): void => {
```

Comment:
I don't think we need to wrap the whole block in a promise. Instead, we could return the promise chain to avoid unnecessary nesting, like:

```ts
return request
  .get()
  .then()
  .then()
 ...
```